config/v1/types_infrastructure.go

```diff
@@ -600,9 +600,300 @@ type OvirtPlatformStatus struct {
+	ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
+
+	// DatacenterFailureDomain failure domain can be only be a region. The vcenter
+	// datacenter is required to be preconfigred and assigned
+	// in the topology. If used the zone would be of type ComputeCluster.
+	DatacenterFailureDomain VSpherePlatformFailureDomainType = "Datacenter"
+)
+
+// VSpherePlatformFailureDomainSpec holds the region and zone failure domain and
+// the vCenter topology of that failure domain.
+type VSpherePlatformFailureDomainSpec struct {
+	// name defines the name of the VSpherePlatformFailureDomainSpec
+	// This name is arbitrary but will be used
+	// in VSpherePlatformDeploymentZone for association.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"name"`
+
+	// region defines a VSpherePlatformFailureDomain which
+	// includes the name of the vCenter tag, the failure domain type
+	// and the name of the vCenter tag category.
+	// +kubebuilder:validation:Required
+	Region VSpherePlatformFailureDomain `json:"region"`
+
+	// zone defines a VSpherePlatformFailureDomain which
+	// includes the name of the vCenter tag, the failure domain type
+	// and the name of the vCenter tag category.
+	// +kubebuilder:validation:Required
+	Zone VSpherePlatformFailureDomain `json:"zone"`
+
+	// Topology describes a given failure domain using vSphere constructs
+	// +kubebuilder:validation:Required
+	Topology VSpherePlatformTopology `json:"topology"`
+}
+
+// VSpherePlatformFailureDomain holds the name of the associated tag, the type
+// of the failure domain, and the vCenter tag category associated with this
+// failure domain.
+type VSpherePlatformFailureDomain struct {
+	// name is the name of the vCenter tag that represents this failure domain
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	Name string `json:"name"`
+
+	// type is the name of the failure domain type, which includes
+	// Datacenter, ComputeCluster and HostGroup
+	// +kubebuilder:validation:Required
+	Type VSpherePlatformFailureDomainType `json:"type"`
+
+	// tagCategory is the category used for the tag
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	TagCategory string `json:"tagCategory"`
+}
+
+// VSpherePlatformTopology holds the required and optional vCenter objects - datacenter,
+// computeCluster, networks, datastore and resourcePool - to provision virtual machines.
+type VSpherePlatformTopology struct {
+	// datacenter is the vCenter datacenter in which virtual machines will be located
+	// and defined as the failure domain.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	Datacenter string `json:"datacenter"`
+
+	// computeCluster as the failure domain
```

Comment:
What does this mean? 

Is this option only valid when type is ComputeCluster? 

Does this need to be a [discrminated union]()https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions?

Reply:
Changed text since we removed HostGroup as a topology option.